firebase-database/src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java

```diff
@@ -1049,6 +1066,32 @@ public void onResponse(Map<String, Object> response) {
         });
   }
 
+  private Task<Object> sendGet(final QuerySpec query, TaskCompletionSource<Object> source) {
+    Map<String, Object> request = new HashMap<String, Object>();
+    request.put(REQUEST_PATH, ConnectionUtils.pathToString(query.path));
+    request.put(REQUEST_QUERIES, query.queryParams);
+    sendAction(
+        REQUEST_ACTION_QUERY,
+        request,
+        new ConnectionRequestCallback() {
+          @Override
+          public void onResponse(Map<String, Object> response) {
+            String status = (String) response.get(REQUEST_STATUS);
+            if (status.equals("ok")) {
```

Comment:
Do we need to special case `404` as well and treat it as a successful response?

Reply:
We would return null if no data matches the query.